vertical-pod-autoscaler/pkg/utils/status/status_updater.go

```diff
@@ -52,7 +53,10 @@ func (su *Updater) Run(stopCh <-chan struct{}) {
 			case <-stopCh:
 				return
 			case <-time.After(su.updateInterval):
-				if err := su.client.UpdateStatus(); err != nil {
+				ctx, cancel := context.WithTimeout(context.Background(), su.updateInterval)
```

Comment:
Note that this will not be repeated each `su.updateInterval` seconds. Rather, it will be repeated each `su.updateInterval` seconds + w/e time it takes to update the status. If you want to run this exactly once each `su.updateInterval` seconds, then you should use a `ticker`. I think `ticker` is also preferred when something has to be executed multiple times, whereas `timer` is preferred when you want to execute something once in the future (`timer.After` will additionally create a new channel on each invocation that has to be garbage collected)

Reply:
Thanks! I addressed the suggestion in e0090f0da2e7b1ef3a20bfac310786d99d8a23fe.